### ISSUE TITLE
Bounding boxes are not drawn even both classification and localization loss is near 0 in RetinaNet: TensorFlow Object Detection API

### ISSUE THREAD

**Issue by shelkesagar29** [tensorflow/models#6273]
### System information
- **What is the top-level directory of the model you are using: ~/models/reserach/object_detection**
- **Have I written custom code : NO**
- **OS Platform and Distribution: Linux Ubuntu 16.04**:
- **TensorFlow installed from source**:
- **TensorFlow version 1.12.0**:
- **Bazel version (if compiling from source)**:
- **CUDA/cuDNN version : 410.86**
- **GPU model and memory : 1080 - 8GB**

### Describe the problem
I am training object detection algorithm using TensorFlow object detection API.

Config File: https://github.com/tensorflow/models/blob/master/research/object_detection/samples/configs/ssd_resnet50_v1_fpn_shared_box_predictor_640x640_coco14_sync.config

Model Name: SSD_resnet50_v1_fpn

After training model using object detection API for around 100000 global steps, total_loss goes to zero. Classification and Localization loss also goes to zero as shown in Tensorboard
![losses_fpn](https://user-images.githubusercontent.com/25697952/53448549-4afa8b00-39dd-11e9-8aaf-58c8e899b640.png)

But evaluation on test images does not perform well and draw any bounding boxes.
Below is the tensorboard for evaluation
![eval_fpn](https://user-images.githubusercontent.com/25697952/53448666-9745cb00-39dd-11e9-89e2-31d47cc6238b.png)

I have checked multiple times for the correctness of data but everyting is right.
In the evaluation tensorboard, images are shown without bounding boxes

### Exact command to reproduce
My Config File: https://drive.google.com/open?id=1oOqvTyZDWtECWPo_5qYgUsOgZMwAVDlk

Train Command
python3 ~/models/research/object_detection/legacy/train.py --logtostderr --train_dir=./fpn/train --pipeline_config_path=./ssd_resnet50_v1_fpn_shared_box_predictor_640x640_coco14_sync.config

Evaluation command
python3 ~/models/research/object_detection/legacy/eval.py --logtostderr --checkpoint_dir=./fpn/train --pipeline_config_path=./ssd_resnet50_v1_fpn_shared_box_predictor_640x640_coco14_sync.config --eval_dir=./fpn/eval

Program to generate tfrecords

https://drive.google.com/open?id=17YwL2KKfqdZgPoHLk85y-D2yOUcXB_AC

Is there anything wrong with generate tf-records script?

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
What is the top-level directory of the model you are using
Have I written custom code
OS Platform and Distribution
GPU model and memory
Exact command to reproduce

**Comment by shelkesagar29** [comment]
I have updated all the fields and added few thing to regenerate the problem

**Comment by shelkesagar29** [comment]
Problem is solved there was some problem with data.

**Comment by Tredence-GitHub** [comment]
I am getting the same problem. No bounding boxes even for the images that I trained the model on.
What was it that solved the problem for you?

**Comment by shelkesagar29** [comment]
If you are getting all losses near zero then there should be an error while creating tfrecords. In my case, there we confusion between [ymin, xmin, ymax, xmax] order from my dataset so after solving that, the model was training properly.

**Comment by shelkesagar29** [comment]
Are you getting loss zero for small dataset? In my case, the small dataset has 9000 training images.

**Comment by prateethvnayak** [comment]
> If you are getting all losses near zero then there should be an error while creating tfrecords. In my case, there we confusion between [ymin, xmin, ymax, xmax] order from my dataset so after solving that, the model was training properly.

Hey.. can you elaborate on this ? I had text annotations for my dataset.. I wrote a tf record using my own logic. The classification loss is working fine. But the Localization loss is zero from the starting.. I'm fine tuning a ssd quantized model. Can you share the tfrecord generation file ?

**Comment by prateethvnayak** [comment]
> If you are getting all losses near zero then there should be an error while creating tfrecords. In my case, there we confusion between [ymin, xmin, ymax, xmax] order from my dataset so after solving that, the model was training properly.

Is there a specific format for the box coordinates while creating the tfrecords ?

**Comment by shelkesagar29** [comment]
If you are using TensorFlow object detection API, expected order is  [ymin, xmin, ymax, xmax]. 
Please check the example in g3doc about preparing your own dataset

https://github.com/tensorflow/models/blob/master/research/object_detection/g3doc/using_your_own_dataset.md

**Comment by prateethvnayak** [comment]
@shelkesagar29 And the naming needs to be exactly the same ? in `tf.example` ? 

```
'image/object/bbox/xmin': dataset_util.float_list_feature(xmins),
'image/object/bbox/xmax': dataset_util.float_list_feature(xmaxs),
'image/object/bbox/ymin': dataset_util.float_list_feature(ymins),
'image/object/bbox/ymax': dataset_util.float_list_feature(ymaxs),
```

**Comment by shelkesagar29** [comment]
yes. Using these keys, values are obtained internally.
Use the exact script with your data

**Comment by SinDongHwan** [comment]
Hi, @shelkesagar29.
I faced the same situation. 

I had trained two case models with a same own dataset.

1) first tried to fine-tuning a model 'faster_resnet101_coco', i reached good mAP.
2) second tried to fine-tuning a model 'ssd_resnet50_v1_fpn', i reached bad mAP.(like mAP graph on your tensorboard)

how did you solve a problem?

**Comment by vanduc103** [comment]
> If you are getting all losses near zero then there should be an error while creating tfrecords. In my case, there we confusion between [ymin, xmin, ymax, xmax] order from my dataset so after solving that, the model was training properly.

I have got the same problem with my localization loss was 0.0 from the beginning but my failure was not in the bounding box order but in the label name in the label_map file. The label name must be exact as in the train/val tfrecord (e.g., in my tfrecord it is Pedestrian_Pedestrian but in my label map file it was Pedestrian). The TF training script does not complain with this mismatching and just do the training. I hope it would be updated in the future. And I log this error for anyone who was suffered by it.
